### ISSUE TITLE
Great works! but how to stream audio and video at the same time with temporally aligned

### ISSUE THREAD

**Issue by TimSeven** [freddyaboulton/gradio-webrtc#7]
It seems to set only one modality (video or audio) at a time?

**Comment by freddyaboulton** [comment]
Yes haven't worked on audio and video simultaneously. What is the use case? Will help design it.

**Comment by Rjx0613** [comment]
> Yes haven't worked on audio and video simultaneously. What is the use case? Will help design it.

When developing digital man's real time chat, it will need streaming audio and frame at same time. Can you design it?

**Comment by Henry-23** [comment]
> > Yes haven't worked on audio and video simultaneously. What is the use case? Will help design it.
> 
> When developing digital man's real time chat, it will need streaming audio and frame at same time. Can you design it?

I have the exact same use case 🤣

**Comment by freddyaboulton** [comment]
How exactly is the video generated in this case? Is there some sample code you can point me to for generating a simultaneous audio/video stream? Not sure how to begin on this one 😄

**Comment by freddyaboulton** [comment]
Should be possible now with version 0.0.28. 

See this demo where audio/video are sent to/from the gemini API simultaneously: https://huggingface.co/spaces/freddyaboulton/gemini-audio-video-chat

https://github.com/user-attachments/assets/cd8806ef-38b3-4f42-855b-84f240443099

In this case, the input video is reflected back to the user. I don't have a good way to generate video on the fly. 

Will close as the basic functionality is in but I expect to follow up on this based on your feedback so please feel free to open new issues.
